xbmc/interfaces/legacy/AddonClass.h

```diff
@@ -126,7 +126,7 @@ namespace XBMCAddon
     void Release() const
 #ifndef XBMC_ADDON_DEBUG_MEMORY
     {
-      long ct = AtomicDecrement((long*)&refs);
+      long ct = AtomicDecrement((std::atomic<long>*)&refs);
```

Comment:
AtomicDecrement can be dropped, operations on atomic declared parameters are atomic